共有ボタンバグ

共有ボタンを押すと以下のようなエラーになる
共有に失敗しました: Error: Call to function 'ExpoSharing.shareAsync' has been rejected.
→ Caused by: java.lang.NoSuchMethodError: No method getFilePermission()Lexpo/modules/interfaces/filesystem/FilePermissionModuleInterface; in class Lexpo/modules/kotlin/AppContext; or its super classes (declaration of 'expo.modules.kotlin.AppContext' appears in /data/app/~~N40pdloHu6X7ql03pBPzBA==/com.eiseikomiya.convert2gabigabi-7VZWFTH6CKnf3LjO5_Z6uA==/base.apk!classes2.dex)

--- a/app/src/screens/MainScreen.tsx
+++ b/app/src/screens/MainScreen.tsx
@@ -380,9 +380,12 @@ const MainScreen = () => {
         return;
       }
       // Copy to cache directory to avoid expo-sharing permission issues on Android
+      // If already in cache, share directly to avoid "same source and destination" error
       const filename = processedImage.split('/').pop() ?? 'shared_image.jpg';
       const cacheUri = `${FileSystem.cacheDirectory}${filename}`;
-      await FileSystem.copyAsync({from: processedImage, to: cacheUri});
+      if (processedImage !== cacheUri) {
+        await FileSystem.copyAsync({from: processedImage, to: cacheUri});
+      }
       await Sharing.shareAsync(cacheUri);
     } catch (err) {
       showError('エラー', `共有に失敗しました: ${String(err)}`);

